lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -47,125 +49,96 @@ const SessionAuth: React.FC<Props> = ({ children, ...props }) => {
-                        setContext({
-                            doesSessionExist: false,
-                            userId: "",
-                            jwtPayload: {},
-                        });
+                    if (props.requireAuth !== true) {
+                        setContext(event.sessionContext);
                     }
                     return;
                 case "UNAUTHORISED":
-                    // If there's onSessionExpired handler, use it without setting state...
-                    if (props.onSessionExpired !== undefined) {
-                        props.onSessionExpired();
-                        return;
-                    }
-
-                    // ...else fallback to default behaviour
-                    if (!hasParentProvider(parentSessionContext)) {
-                        setContext({
-                            doesSessionExist: false,
-                            userId: "",
-                            jwtPayload: {},
-                        });
+                    if (props.requireAuth === true) {
+                        if (props.onSessionExpired !== undefined) {
+                            props.onSessionExpired();
+                        } else {
+                            props.redirectToLogin();
+                        }
+                    } else {
+                        setContext(event.sessionContext);
+                        if (props.onSessionExpired !== undefined) {
+                            props.onSessionExpired();
+                        }
                     }
                     return;
             }
-        },
-        [parentSessionContext, context, props]
-    );
-
-    useEffect(() => {
-        // If there's a parent provider, it already listens for events, so we don't have to
-        if (hasParentProvider(parentSessionContext)) {
-            setContext(parentSessionContext);
-            return;
-        }
-    }, [parentSessionContext]);
-
-    // Read and set the current state
-    useEffect(() => {
-        if (hasParentProvider(parentSessionContext)) {
-            return;
-        }
-
-        if (context === undefined) {
-            setInitialContext();
         }
-    }, [context, parentSessionContext, setInitialContext]);
 
-    useEffect(() => {
         // we return here cause addEventListener returns a function that removes
         // the listener, and this function will be called by useEffect when
         // onHandleEvent changes or if the component is unmounting.
         return session.current.addEventListener(onHandleEvent);
-    }, [onHandleEvent]);
-
-    useEffect(() => {
-        if (context === undefined) {
-            return;
-        }
-
-        // If the session doesn't exist and we require auth, redirect to login
-        if (context.doesSessionExist === false && props.requireAuth === true) {
-            props.redirectToLogin();
-            return;
-        }
-    }, [context, props]);
+    }, []);
```

Comment:
If we don't pass dependencies here, we're going to have a stale handler (because props may change). This is fine, as this would be the behaviour of `componentDidMount`, but needs to be documented.

Reply:
I have a comment `// subscribe to events on mount` on top of the `useEffect`. We don't need to document it in the docs explicitly.. serves no purpose. You can add more comments in there if you like...

Reply:
- This is meant to be run on mount, but can also run when `props` change (as long as the older event listening is unsubscribed).
- Adding `props` to the dependency array should fix it.